src/corva/service/cache_sdk.py

```diff
@@ -0,0 +1,159 @@
+import datetime
+import warnings
+from typing import List, Optional, Protocol, overload
+
+import fakeredis
+import redis
+
+from corva import cache_adapter
+
+
+class UserCacheSdkProtocol(Protocol):
+    def set(self, key: str, value: str, ttl: int = ...) -> None:
+        ...
+
+    def get(self, key: str) -> Optional[str]:
+        ...
+
+    # TODO: remove asterisk in v2 - it was added for backward compatibility
+    def delete(self, *, key: str) -> None:
+        ...
+
+
+class UserRedisSdk:
```

Comment:
User gets this intefrace implementation (adapter). Which has all old methods (with deprecation warnings) + new methods